### PR TITLE
MGDOBR-731: Update processors filters

### PR DESCRIPTION
**Pull request by vpellegrino**
**Task:** https://issues.redhat.com/browse/MGDOBR-731
image
image
**Description:** We also want to support `string in` and `number in` filters on the UI. This is an implementation to make it possible: the filters that have multiple values can be set with a comma-separated string (`1,2,3`, `hi,how,are,you`).

From the _usability_ point of view, we will show this approach to the UX team for collecting feedback that can eventually be accomplished in a separate task.
Like for other filters, there is no validation in place. Instead, an "optimistic" approach has been followed: each "valid" filter item is kept, and the others are discarded.

### CODE REVIEW

File: src/app/Processor/ProcessorEdit/ProcessorEdit.tsx [hunk 1]
```diff
@@ -116,6 +121,40 @@ const ProcessorEdit = (props: ProcessorEditProps): JSX.Element => {
...
+            .filter(multipleValuesPredicate);
+
+          return {
+            key: filter.key,
+            type: filter.type,
+            values,
+          };
+        }
+        return filter;
+      });
```
Comment (jomarko) [inline src/app/Processor/ProcessorEdit/ProcessorEdit.tsx:156]: Hello Valentino, would you help me/guide, how to measure coverage of this? It seems to me we miss some unit tests. But I may be wrong, and I will learn at least something new.

**Review by jomarko — COMMENTED** [review]
Please, do you have any pointers to backend `sandbox`, so we are sure such filters exists there? I do not recall them form times I followed some RHOSE getting started guide.

**Comment by kelvah** [comment]
> Please, do you have any pointers to backend `sandbox`, so we are sure such filters exists there? I do not recall them form times I followed some RHOSE getting started guide.

Hi @jomarko, the new filters covered here were introduced with [this PR](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/584). You can find their documentation in the (updated) [filters readme](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/FILTERS.md#stringin).

File: src/app/Processor/ProcessorEdit/ProcessorEdit.tsx (hunk 1 above)
Comment (kelvah) [reply src/app/Processor/ProcessorEdit/ProcessorEdit.tsx]: @jomarko I've added a test. Let me know if it's more clear now.

**Comment by kelvah** [comment]
I've just realized we also have to manage an array of values for `StringContains` and `StringBeginsWith`...

File: src/app/Processor/ProcessorDetail/ProcessorDetail.tsx [hunk 2]
```diff
@@ -123,8 +124,15 @@ const ProcessorDetail = (props: ProcessorDetailProps): JSX.Element => {
...
                         <Td>{filter.key}</Td>
-                        <Td>{t(`processor.${filter.type}`)}</Td>
-                        <Td>{filter.value}</Td>
+                        <Td>
+                          {t(
+                            `processor.${
+                              filter.type.charAt(0).toLowerCase() +
+                              filter.type.slice(1)
+                            }`
+                          )}
```
[the remark this answers is not in the document]
Comment (vpellegrino) [reply src/app/Processor/ProcessorDetail/ProcessorDetail.tsx]: Changed with https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/72/commits/c48572028562d60d3ec4a74f0aa7b66adb6f789c

**Comment by vpellegrino** [comment]
@jomarko @kelvah any point on the PR that needs my attention?

File: src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx [hunk 3]
```diff
@@ -52,27 +58,62 @@ const FiltersEdit = (props: FiltersEditProps): JSX.Element => {
...
+          className="pf-c-form__group-label-help"
+        >
+          <HelpIcon noVerticalAlign={true} />
+        </button>
+      </Popover>
+    ) : undefined;
+
+  const getOptionalPlaceholder = (filter: EventFilter): string | undefined => {
+    if (filter.type === FilterType.STRING_IN) {
+      return "hello,world,how,are,you";
```
Comment (jomarko) [inline src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx:109]: how is user intended to put values like:
- "Hello World"
- "Jozef's query"

I mean should we demonstrate non single word strings in the help? Are such supported by backend?

**Review by jomarko — COMMENTED** [review]
One question form my @vpellegrino

**Review by kelvah — APPROVED** [review]
LGTM

File: src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx (hunk 3 above)
Comment (vpellegrino) [reply src/app/Processor/ProcessorEdit/FiltersEdit/FiltersEdit.tsx:109]: Input string is comma separated, so whatever you put between will be considered a filter item.
So, for your example: "Hello World, Jozef's query".
For the backend I think spaces are perfectly fine. However you can try it out with the OpenApi web interface

**Pull request merged**